aws/external/resolve.go

```diff
@@ -136,3 +138,52 @@ func ResolveEndpointResolverFunc(cfg *aws.Config, configs Configs) error {
 
 	return nil
 }
+
+// ResolveDefaultRegion extracts the first instance of a default region and sets `aws.Config.Region` to the default
+// region if region had not been resolved from other sources.
+func ResolveDefaultRegion(cfg *aws.Config, configs Configs) error {
+	if len(cfg.Region) > 0 {
+		return nil
+	}
+
+	region, found, err := GetDefaultRegion(configs)
+	if err != nil {
+		return nil
+	}
```

Comment:
should this resolver allow the GetDefaultRegion error to be retruned?

Reply:
Ah yes it should good catch!